packages/expo-facebook/android/src/main/java/expo/modules/facebook/FacebookModule.kt

```diff
@@ -0,0 +1,342 @@
+    private const val ERR_FACEBOOK_MISCONFIGURED = "ERR_FACEBOOK_MISCONFIGURED"
+    private const val ERR_FACEBOOK_LOGIN = "ERR_FACEBOOK_LOGIN"
+    private const val PUSH_PAYLOAD_KEY = "fb_push_payload"
+    private const val PUSH_PAYLOAD_CAMPAIGN_KEY = "campaign"
+  }
+
+  private inline fun <reified T> moduleRegistry() = moduleRegistryDelegate.getFromModuleRegistry<T>()
+
+  override fun getName(): String {
+    return "ExponentFacebook"
+  }
+
+  override fun onCreate(moduleRegistry: ModuleRegistry) {
+    moduleRegistryDelegate.onCreate(moduleRegistry)
+    uIManager.registerActivityEventListener(this)
+  }
+
+  override fun onActivityResult(activity: Activity, requestCode: Int, resultCode: Int, data: Intent) {
+    callbackManager.onActivityResult(requestCode, resultCode, data)
+  }
+
+  override fun onNewIntent(intent: Intent) {
+    // do nothing
+  }
+
+  private fun bundleWithNullValuesAsStrings(parameters: ReadableArguments?): Bundle {
+    return Bundle().apply {
+      if (parameters != null) {
+        for (key in parameters.keys()) {
+          val value = parameters[key]
+          if (value == null) {
+            putString(key, "null")
+          } else if (value is String) {
+            putString(key, value)
+          } else if (value is Int) {
+            putInt(key, value)
+          } else if (value is Double) {
+            putDouble(key, value)
+          } else if (value is Long) {
+            putLong(key, value)
+          }
+        }
+      }
+    }
+  }
+
+  @ExpoMethod
+  fun setAutoLogAppEventsEnabledAsync(enabled: Boolean, promise: Promise) {
+    FacebookSdk.setAutoLogAppEventsEnabled(enabled)
+    promise.resolve(null)
+  }
+
+  @ExpoMethod
+  fun setAdvertiserIDCollectionEnabledAsync(enabled: Boolean, promise: Promise) {
+    FacebookSdk.setAdvertiserIDCollectionEnabled(enabled)
+    promise.resolve(null)
+  }
+
+  @ExpoMethod
+  fun getAuthenticationCredentialAsync(promise: Promise) {
+    val accessToken = AccessToken.getCurrentAccessToken()
+    promise.resolve(accessTokenToBundle(accessToken))
+  }
+
+  private fun accessTokenToBundle(accessToken: AccessToken?): Bundle? {
+    return if (accessToken != null) {
+      Bundle().apply {
+        putString("token", accessToken.token)
+        putString("userId", accessToken.userId)
+        putString("appId", accessToken.applicationId)
+        putStringArrayList("permissions", ArrayList(accessToken.permissions))
+        putStringArrayList("declinedPermissions", ArrayList(accessToken.declinedPermissions))
+        putStringArrayList("expiredPermissions", ArrayList(accessToken.expiredPermissions))
+        putDouble("expirationDate", accessToken.expires.time.toDouble())
+        putDouble("dataAccessExpirationDate", accessToken.dataAccessExpirationTime.time.toDouble())
+        putDouble("refreshDate", accessToken.lastRefresh.time.toDouble())
+        putString("tokenSource", accessToken.source.name)
+      }
+    } else {
+      null
+    }
```

Comment:
Up to you - this way you can get rid of that lonely `null` at line 122 😉 
```suggestion
    return accessToken?.let {
      Bundle().apply {
        putString("token", it.token)
        putString("userId", it.userId)
        putString("appId", it.applicationId)
        putStringArrayList("permissions", ArrayList(it.permissions))
        putStringArrayList("declinedPermissions", ArrayList(it.declinedPermissions))
        putStringArrayList("expiredPermissions", ArrayList(it.expiredPermissions))
        putDouble("expirationDate", it.expires.time.toDouble())
        putDouble("dataAccessExpirationDate", it.dataAccessExpirationTime.time.toDouble())
        putDouble("refreshDate", it.lastRefresh.time.toDouble())
        putString("tokenSource", it.source.name)
      }
    }
```